rift/node.py

```diff
@@ -3046,18 +3053,36 @@ def spf_install_routes_in_rib(self, spf_direction):
                 prefix = dest_key
                 if prefix.ipv4prefix is not None:
                     next_hops = dest.ipv4_next_hops
-                    route_table = self._ipv4_rib
+                    if prefix not in all_next_hops_ipv4:
+                        all_next_hops_ipv4[prefix] = {}
+                    all_next_hops_ipv4[prefix][dest.dest_type] = next_hops
                 else:
                     assert prefix.ipv6prefix is not None
                     next_hops = dest.ipv6_next_hops
-                    route_table = self._ipv6_rib
-                if next_hops:
-                    rte = route.Route(prefix, owner, next_hops)
-                    route_table.put_route(rte)
+                    if prefix not in all_next_hops_ipv6:
+                        all_next_hops_ipv6[prefix] = {}
+                    all_next_hops_ipv6[prefix][dest.dest_type] = next_hops
+
+        self._install_rib_routes(all_next_hops_ipv4, owner, self._ipv4_rib)
+        self._install_rib_routes(all_next_hops_ipv6, owner, self._ipv6_rib)
 
         self._ipv4_rib.del_stale_routes()
         self._ipv6_rib.del_stale_routes()
 
+    @staticmethod
+    def _install_rib_routes(destinations_next_hops, owner, route_family_rib):
```

Comment:
@brunorijsman this is a new method to install both positive and negative next hops after SPF computation